app/src/lib/logging/logger.ts

```diff
@@ -1,42 +1,43 @@
+import { app } from 'electron'
 import * as winston from 'winston'
 require('winston-daily-rotate-file')
 
 import * as Fs from 'fs-extra'
 import * as Path from 'path'
 
-import { ElectronConsole } from './electron-console'
-
 export const LogFolder = 'logs'
 
-export interface ILogger {
-  readonly debug: (message: string) => void
-  readonly info: (message: string) => void
-  readonly error: (message: string, error?: Error) => void
-}
-
 /** resolve the log file location based on the current environment */
 export function getLogFilePath(directory: string): string {
   const environment = process.env.NODE_ENV || 'production'
   const fileName = `desktop.${environment}.log`
   return Path.join(directory, fileName)
 }
 
-/** wireup the file and console loggers */
-function create(filename: string) {
+/**
+ * Initializes winston and returns a subset of the available log level
+ * methods (debug, info, error). This method should only be called once
+ * during an application's lifetime.
+ * 
+ * @param path The path where to write log files. This path will have
+ *             the current date prepended to the basename part of the
+ *             path such that passing a path '/logs/foo' will end up
+ *             writing to '/logs/2017-05-17.foo'
+ */
+function initializeWinston(path: string): winston.LogMethod {
   const fileLogger = new winston.transports.DailyRotateFile({
-    filename,
-    humanReadableUnhandledException: true,
-    handleExceptions: true,
```

Comment:
This makes winston attach its own `uncaughtException` handler which is redundant since we already do